### PR TITLE
Migrated project from Gradle to Maven

### PR DESCRIPTION
**Pull request by ghost**
This migrates the benchmark from a Maven project contained within a Gradle wrapper to a true Maven project.

I've also explicitly set versions for 2 of the dependencies, as Maven was warning me that those were missing.

### CODE REVIEW

File: diffblue.yml [hunk 1]
```diff
@@ -1,2 +1,2 @@
-buildCmd: 'gradle clean assemble'
-testCmd: 'gradle test -x compileJava'
+buildCmd: 'mvn compile'
```
Comment (rjmunro) [inline diffblue.yml:1]: Could you not avoid the rest of this PR by doing something like:

```yaml
buildCmd: 'cd url-detector; mvn compile'
testCmd: 'cd url-detector; mvn test'
```

**Review by rjmunro — COMMENTED** [review]
We really should be aiming to get projects working as they are, only adding a diffblue.yml if necessary.

If diffblue can't be made to work on a certain project structure, then it should be seen as a bug in diffblue, and we should fix it if that structure is frequently found in the field.

File: diffblue.yml (hunk 1 above)
Comment (ghost) [reply diffblue.yml:1]: True, but i'd prefer to strip out the Gradle wrapper completely, as its unnecessary from the point of view of our internal work using the repository etc. 

The project still works fine using Gradle as originally forked. If you'd prefer we kept the original & my true Maven version as separate branches, i'm fine with that.

**Comment by ghost** [comment]
I've migrated the changes into a more accurately-named branch. I won't go ahead with merging into our master for now, due to the above concerns, but its there if you want to use it.

**Pull request closed without merging**